### PR TITLE
(DOCSP-11379) Update Titles and Heading page

### PR DESCRIPTION
**Pull request by atsansone**
@andf-mongodb : [Staged](https://docs-mongodbcom-staging.corp.mongodb.com/meta/docsworker-xlarge/DOCSP-11379/style-guide/style/titles-and-headings.html)
---
[Builds without errors on `style-guide-v3`.](https://workerpool-boxgs.mongodbstitch.com/pages/job.html?jobId=5f209a432f9a5cfb5b9c0ea3)

### CODE REVIEW

File: source/style-guide/style/titles-and-headings.txt [hunk 1]
```diff
@@ -276,9 +279,11 @@ contents (TOC) for a collection of content:
...
    Don't include a link to an outside resource in the TOC.
 
--  The text of a TOC entry must match the text of the title or heading
-   to which it links. If the link needs to be shorter, revise the
-   title or heading to be shorter.
+-  The text of a TOC entry should match the text of the title or
+   heading to which it links. If the link needs to be shorter, consider
+   shortening the title or heading or providing a more concise title or
+   heading for the TOC. An alternate TOC title or heading should convey
+   the same intent of the full title or heading.
```
Comment (andf-mongodb) [inline source/style-guide/style/titles-and-headings.txt:286]: Might this be a good place to bring up renaming anchor links i.e. `:ref:` links where I change the name of the link away from its default header text? Example:

This FAQ item is very helpful:

https://docs.mongodb.com/master/faq/concurrency/#what-locks-are-taken-by-some-common-client-operations

But the question format isn't really appropriate for use in reference material. So I would probably do something like this:

> See the :ref:`list of common client operations requiring locks`<that_ref> for more information.

Comment (andf-mongodb) [reply source/style-guide/style/titles-and-headings.txt:286]: (this is for refs, not the TOC, so different application, but similar approach.

Comment (atsansone) [reply source/style-guide/style/titles-and-headings.txt:286]: Added an example.

File: source/style-guide/style/titles-and-headings.txt [hunk 2]
```diff
@@ -285,5 +285,22 @@ contents (TOC) for a collection of content:
...
+            :titlesonly:
+
+            Install the Operator </installation>
+            Deploy Ops Manager Resources </om-resources>
+            Deploy MongoDB Database Resources </mdb-resources>
+            /tutorial/modify-resource-image
+            /reference
+            Release Notes </release-notes>
+            MongoDB Community Kubernetes Operator <https://github.com/mongodb/mongodb-kubernetes-operator>
+
```
Comment (andf-mongodb) [inline source/style-guide/style/titles-and-headings.txt:304]: Ah, sorry, I didn't mean as a TOC entry, though this example is indeed useful.

I meant only the `:ref:` directive, where renaming away from the default heading link is common practice in Serverland. Ping me if you want to talk more, of feel free to disregard if `:ref:` doesn't need a callout here.

Comment (atsansone) [reply source/style-guide/style/titles-and-headings.txt:304]: Not here. I'll cover it in the markup guide.

**Pull request merged**